README.md

```diff
@@ -5,25 +5,78 @@ Origin Kubernetes
 [![GoDoc](https://godoc.org/github.com/openshift/origin?status.png)](https://godoc.org/github.com/openshift/origin)
 [![Licensed under Apache License version 2.0](https://img.shields.io/github/license/openshift/origin.svg?maxAge=2592000)](https://www.apache.org/licenses/LICENSE-2.0)
 
-This is the core Kubernetes tracking repo for OKD. For more about OKD, see https://github.com/openshift/okd.
+OpenShift's E2E test suite (`openshift-tests`) is maintained in this
+repo.
 
+This repo was previously the core Kubernetes tracking repo for
+[OKD](https://github.com/openshift/okd).
 
-Contributing
-------------
+## Hyperkube maintenance
 
-All contributions are welcome - OKD uses the Apache 2 license and does not require any contributor agreement to submit patches.  Please open issues for any bugs or problems you encounter, ask questions on the OpenShift IRC channel (#openshift-dev on freenode), or get involved in the [Kubernetes project](https://github.com/kubernetes/kubernetes) at the container runtime layer.
+### Release >= 4.6
 
-See [HACKING.md](https://github.com/openshift/origin/blob/master/HACKING.md) for more details on developing on Origin including how different tests are setup.
+The hyperkube binaries, previously maintained in this repo, are now
+maintained in
+[openshift/kubernetes](https://github.com/openshift/kubernetes). Persistent
+carries and backports should be proposed to that repo. Once changes
+land in `openshift/kubernetes`, they can be vendored by this repo as
+needed.
 
+### Releases < 4.6
 
-Security Response
------------------
-If you've found a security issue that you'd like to disclose confidentially
-please contact Red Hat's Product Security team. Details at
-https://access.redhat.com/security/team/contact
+Releases prior to 4.6 continue to maintain hyperkube in the origin
+repo in the `release-4.x` branches. Persistent carries and backports
+for those branches should continue to be submitted directly to origin.
 
+## Vendoring k8s.io/kubernetes
 
-License
--------
+This repo vendors `k8s.io/kubernetes` and some of its staging repos
+(e.g. `k8s.io/api`) from our
+[openshift/kubernetes](https://github.com/openshift/kubernetes)
+fork. Upstream staging repos are used where possible, but some tests
+depends on functionality only present in the fork.
 
-OKD is licensed under the [Apache License, Version 2.0](http://www.apache.org/licenses/).
+### Updating k8s.io/kubernetes vendoring
+
+When bumping the version of `openshift/kubernetes` vendored by this
+repo, it's necessary to update the version of staging repos that are
+also sourced (via a subpath) from `openshift/kubernetes`. A helper
```

Comment:
this sentence confuses me. How would I "update the version of staging repos" ?

Reply:
I think it is super important to say which o/k branch we vendor and that all upstream patches must go there first via PR.

Reply:
I say as much beginning on line 20. This section is specific to how to update the vendoring.

Reply:
> all upstream patches must go there first via PR.

I say as much beginning on line 20. This section is specific to how to update the vendoring.

Reply:
line 20 is about hyperkube, not about origin. Do I miss anything? Which branch do we vendor here in origin master?

Reply:
The branch names should match between the 2 repos (i.e. master branch of origin should vendor master branch of openshift/kubernetes). I've updated the docs to hopefully provide clarity, PTAL.